xclim/indices/_agro.py

```diff
@@ -589,6 +595,225 @@ def water_budget(
     return out
 
 
+@declare_units(
+    pr="[precipitation]",
+    pr_cal="[precipitation]",
+)
+def standardized_precipitation_index(
+    pr: xarray.DataArray,
+    pr_cal: xarray.DataArray,
+    freq: str = "MS",
+    window: int = 1,
+    dist: str = "gamma",
+    method: str = "ML",
+) -> xarray.DataArray:
+    r"""Standardized Precipitation Index (SPI).
+
+    Parameters
+    ----------
+    pr : xarray.DataArray
+      Daily precipitation.
+    pr_cal : xarray.DataArray
+      Daily precipitation used for calibration. Usually this is a temporal subset of `pr` over some reference period.
+    freq : str
+      Resampling frequency. A monthly or daily frequency is expected.
+    window : int
+      Averaging window length relative to the resampling frequency. For example, if `freq="MS"`, i.e. a monthly resampling, the window
+      is an integer number of months.
+    dist : str
+      Name of the univariate distribution, such as `beta`, `expon`, `genextreme`, `gamma`, `gumbel_r`, `lognorm`, `norm`
```

Comment:
```suggestion
    dist : {'gamma'}
      Name of the univariate distribution, only `gamma` is currently implemented
```
We shouldn't list methods we don't support I think.